File: core/src/main/java/com/sequenceiq/cloudbreak/converter/AccountPreferencesToJsonConverter.java
package com.sequenceiq.cloudbreak.converter;

import org.springframework.stereotype.Component;

import com.sequenceiq.cloudbreak.api.model.AccountPreferencesJson;
import com.sequenceiq.cloudbreak.domain.AccountPreferences;


@Component
public class AccountPreferencesToJsonConverter extends AbstractConversionServiceAwareConverter<AccountPreferences, AccountPreferencesJson> {
    private static final long HOUR_IN_MS = 3600000L;

    private static final long ZERO = 0L;

    @Override
    public AccountPreferencesJson convert(AccountPreferences source) {
        AccountPreferencesJson json = new AccountPreferencesJson();
        json.setMaxNumberOfClusters(source.getMaxNumberOfClusters());
        json.setMaxNumberOfNodesPerCluster(source.getMaxNumberOfNodesPerCluster());
        json.setAllowedInstanceTypes(source.getAllowedInstanceTypes());
        long clusterTimeToLive = source.getClusterTimeToLive() == ZERO ? ZERO : source.getClusterTimeToLive() / HOUR_IN_MS;
        json.setClusterTimeToLive(clusterTimeToLive);
        long userTimeToLive = source.getUserTimeToLive() == ZERO ? ZERO : source.getUserTimeToLive() / HOUR_IN_MS;
        json.setUserTimeToLive(userTimeToLive);
        json.setMaxNumberOfClustersPerUser(source.getMaxNumberOfClustersPerUser());
        json.setPlatforms(source.getPlatforms());
        return json;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/AccountPreferencesToJsonConverter.kt
package com.sequenceiq.cloudbreak.converter

import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.domain.AccountPreferences
import com.sequenceiq.cloudbreak.api.model.AccountPreferencesJson


@Component
class AccountPreferencesToJsonConverter : AbstractConversionServiceAwareConverter<AccountPreferences, AccountPreferencesJson>() {

    override fun convert(source: AccountPreferences): AccountPreferencesJson {
        val json = AccountPreferencesJson()
        json.maxNumberOfClusters = source.maxNumberOfClusters
        json.maxNumberOfNodesPerCluster = source.maxNumberOfNodesPerCluster
        json.allowedInstanceTypes = source.allowedInstanceTypes
        val clusterTimeToLive = if (source.clusterTimeToLive === ZERO) ZERO else source.clusterTimeToLive!! / HOUR_IN_MS
        json.clusterTimeToLive = clusterTimeToLive
        val userTimeToLive = if (source.userTimeToLive === ZERO) ZERO else source.userTimeToLive!! / HOUR_IN_MS
        json.userTimeToLive = userTimeToLive
        json.maxNumberOfClustersPerUser = source.maxNumberOfClustersPerUser
        json.platforms = source.platforms
        return json
    }

    companion object {
        private val HOUR_IN_MS = 3600000L
        private val ZERO = 0L
    }
}
